frontend/packages/dev-console/src/components/topology/TopologyDataRetriever.tsx

```diff
@@ -1,31 +1,120 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
 import {
   useK8sWatchResources,
   WatchK8sResources,
 } from '@console/internal/components/utils/k8s-watch-hook';
-import { TopologyResourcesObject } from './topology-types';
-import { TopologyDataRenderer, TopologyDataRendererProps } from './TopologyDataRenderer';
+import { PROMETHEUS_TENANCY_BASE_PATH } from '@console/internal/components/graphs';
+import { useURLPoll } from '@console/internal/components/utils/url-poll-hook';
+import { Alerts, PrometheusRulesResponse } from '@console/internal/components/monitoring/types';
+import { getAlertsAndRules } from '@console/internal/components/monitoring/utils';
+import { RootState } from '@console/internal/redux';
+import { TopologyResourcesObject, TrafficData } from './topology-types';
+import ModelContext, { ExtensibleModel } from './data-transforms/ModelContext';
+import { baseDataModelGetter } from './data-transforms';
+import { getFilterById, SHOW_GROUPS_FILTER_ID, useDisplayFilters } from './filters';
 
-export type TopologyDataRetrieverProps = Omit<TopologyDataRendererProps, 'resources'> & {
-  resourcesList: WatchK8sResources<any>;
+interface StateProps {
+  kindsInFlight: boolean;
+}
+
+export type TopologyDataRetrieverProps = {
+  trafficData?: TrafficData;
 };
 
-export const TopologyDataRetriever: React.FC<TopologyDataRetrieverProps> = ({
-  render,
-  resourcesList,
-  namespace,
-  showGraphView,
+const POLL_DELAY = 15 * 1000;
+
+export const ConnectedTopologyDataRetriever: React.FC<TopologyDataRetrieverProps & StateProps> = ({
+  kindsInFlight,
   trafficData,
 }) => {
+  const dataModelContext = React.useContext<ExtensibleModel>(ModelContext);
+  const { namespace } = dataModelContext;
+  const filters = useDisplayFilters();
+  const showGroups = getFilterById(SHOW_GROUPS_FILTER_ID, filters)?.value ?? true;
+  const resourcesList = React.useMemo<WatchK8sResources<any>>(
+    () => (dataModelContext.extensionsLoaded ? dataModelContext.watchedResources : {}),
+    [dataModelContext.extensionsLoaded, dataModelContext.watchedResources],
+  );
+
   const resources = useK8sWatchResources<TopologyResourcesObject>(resourcesList);
 
-  return (
-    <TopologyDataRenderer
-      render={render}
-      namespace={namespace}
-      showGraphView={showGraphView}
-      resources={resources}
-      trafficData={trafficData}
-    />
+  const url = PROMETHEUS_TENANCY_BASE_PATH
+    ? `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`
+    : null;
+  const [response, error, loading] = useURLPoll<PrometheusRulesResponse>(
+    url,
+    POLL_DELAY,
+    namespace,
   );
+
+  const monitoringAlerts: Alerts = React.useMemo(() => {
+    const { alerts } = getAlertsAndRules(response?.data);
+    return { data: alerts, loaded: !loading, loadError: error };
+  }, [response, error, loading]);
```

Comment:
Can you extract this into a separate hook?
Later on we should look to contribute this through extensions somehow.

Reply:
I was planning to do that in a separate PR